spyder_unittest/unittestplugin.py

```diff
@@ -24,16 +26,67 @@
 _ = get_translation("unittest", dirname="spyder_unittest")
 
 
-class UnitTestPlugin(UnitTestWidget, SpyderPluginMixin):
-    """Unit testing."""
+class UnitTestWidgetInSpyder(UnitTestWidget):
+    """
+    Unit test widget for use inside Spyder.
+
+    This class overrides some functions in `UnitTestWidget` to provide better
+    integration with Spyder.
+    """
+
+    def __init__(self, main):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        main : MainWindow
+            Spyder main window
+        """
+        UnitTestWidget.__init__(self, parent=main)
+        self.main = main
+
+    def get_pythonpath(self):
+        """
+        Return directories to be added to the Python path.
+
+        Use Python path from Spyder. Overrides function in base class.
+
+        Returns
+        -------
+        list of str
+        """
+        return self.main.get_spyder_pythonpath()
+
+    def get_default_config(self):
+        """
+        Return configuration which is proposed when current config is invalid.
+
+        Propose to use directory of current file as working directory for
+        testing.
+        """
+        filename = self.main.editor.get_current_filename()
+        dirname = os.path.dirname(filename)
+        return Config(wdir=dirname)
+
+
+class UnitTestPlugin(SpyderPluginWidget):
+    """Spyder plugin for unit testing."""
 
     CONF_SECTION = 'unittest'
-    edit_goto = Signal(str, int, str)
 
     def __init__(self, parent=None):
-        """Unit testing."""
-        UnitTestWidget.__init__(self, parent=parent)
-        SpyderPluginMixin.__init__(self, parent)
+        """Initialize plugin and corresponding widget."""
+        if spyder.version_info[0] < 4 and PYQT5:
```

Comment:
I don't see how I can just use
```
SpyderPluginWidget.__init__(self, parent)
```
with Spyder 3. 

For instance, in the Help plugin, if I remove the `main` argument in https://github.com/spyder-ide/spyder/blob/3.x/spyder/plugins/help.py#L338 on the `3.x` branch, then `self.main` is not set to the main window leading to an error in `register_plugin()`.